README.md

```diff
@@ -796,6 +796,143 @@ The `profunctor` method takes two arguments:
 
 3. `promap` must return a value of the same Profunctor
 
+## Types
+
+In Fantasy Land, types are specified by an object containing a catamorphic
+method `cata`. `cata` has an arity matching the number of constructor functions
```

Comment:
I think it might be better to use different names for each type, instead of using one for all, as for example if sanctuary or ramda wanted to implement it, i don't see what should be a type for `cata`.
I would go with `either` for `Either`, `maybe` for `Maybe` and `runIdentity for `Identity`.

This way one could define maybe instance for Array for example, which might be useful in some cases.


Also regarding this general introduction of the `# Types` I think this part is a bit confusing, which as confusing as type of `cata` could be :D

If we instead go with special names for specific types, then we would have more precise description for each type.

Reply:
> I would go with `either` for `Either`, `maybe` for `Maybe` and `runIdentity` for `Identity`

> This way one could define maybe instance for Array for example, which might be useful in some cases.

See #280 for an alternate proposal. It is identical to what you propose and would allow the variety of polymorphism you're talking about.